[2024-11-19T19:36:45.170+0100] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-19T19:36:45.203+0100] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_pipeline_to_snowflake.run_data_transformation manual__2024-11-19T00:21:47.385967+00:00 [queued]>
[2024-11-19T19:36:45.219+0100] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_pipeline_to_snowflake.run_data_transformation manual__2024-11-19T00:21:47.385967+00:00 [queued]>
[2024-11-19T19:36:45.222+0100] {taskinstance.py:2865} INFO - Starting attempt 17 of 17
[2024-11-19T19:36:45.271+0100] {taskinstance.py:2888} INFO - Executing <Task(PythonOperator): run_data_transformation> on 2024-11-19 00:21:47.385967+00:00
[2024-11-19T19:36:45.279+0100] {standard_task_runner.py:72} INFO - Started process 238762 to run task
[2024-11-19T19:36:45.287+0100] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'etl_pipeline_to_snowflake', 'run_data_transformation', 'manual__2024-11-19T00:21:47.385967+00:00', '--job-id', '70', '--raw', '--subdir', 'DAGS_FOLDER/staging_data_dag.py', '--cfg-path', '/tmp/tmpvstt57fy']
[2024-11-19T19:36:45.291+0100] {standard_task_runner.py:105} INFO - Job 70: Subtask run_data_transformation
[2024-11-19T19:36:45.410+0100] {task_command.py:467} INFO - Running <TaskInstance: etl_pipeline_to_snowflake.run_data_transformation manual__2024-11-19T00:21:47.385967+00:00 [running]> on host zainab.
[2024-11-19T19:36:45.565+0100] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='etl_pipeline_to_snowflake' AIRFLOW_CTX_TASK_ID='run_data_transformation' AIRFLOW_CTX_EXECUTION_DATE='2024-11-19T00:21:47.385967+00:00' AIRFLOW_CTX_TRY_NUMBER='17' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-19T00:21:47.385967+00:00'
[2024-11-19T19:36:45.571+0100] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-11-19T19:36:45.574+0100] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-11-19T19:36:45.577+0100] {logging_mixin.py:190} INFO - Current task name:run_data_transformation state:running start_date:2024-11-19 18:36:45.206493+00:00
[2024-11-19T19:36:45.581+0100] {logging_mixin.py:190} INFO - Dag name:etl_pipeline_to_snowflake and current dag run status:running
[2024-11-19T19:36:45.585+0100] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-19T19:36:46.231+0100] {transform.py:88} ERROR - Error in transformation: sequence item 0: expected str instance, NoneType found
[2024-11-19T19:36:46.235+0100] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/capstone_project/airflow/dags/scripts/transform.py", line 79, in run_transformation
    transformed_data = transform_data_s3(data)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/capstone_project/airflow/dags/scripts/transform.py", line 64, in transform_data_s3
    "Languages": df["languages"].apply(lambda x: ", ".join(x.values()) if isinstance(x, dict) else None),
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/pandas/core/series.py", line 4764, in apply
    ).apply()
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/pandas/core/apply.py", line 1209, in apply
    return self.apply_standard()
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/pandas/core/apply.py", line 1289, in apply_standard
    mapped = obj._map_values(
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/pandas/core/base.py", line 921, in _map_values
    return algorithms.map_array(arr, mapper, na_action=na_action, convert=convert)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/pandas/core/algorithms.py", line 1814, in map_array
    return lib.map_infer(values, mapper, convert=convert)
  File "lib.pyx", line 2926, in pandas._libs.lib.map_infer
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/capstone_project/airflow/dags/scripts/transform.py", line 64, in <lambda>
    "Languages": df["languages"].apply(lambda x: ", ".join(x.values()) if isinstance(x, dict) else None),
TypeError: sequence item 0: expected str instance, NoneType found
[2024-11-19T19:36:46.321+0100] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-11-19T19:36:46.323+0100] {logging_mixin.py:190} INFO - Task start:2024-11-19 18:36:45.206493+00:00 end:2024-11-19 18:36:46.321308+00:00 duration:1.114815
[2024-11-19T19:36:46.325+0100] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): run_data_transformation> dag:<DAG: etl_pipeline_to_snowflake> dagrun:<DagRun etl_pipeline_to_snowflake @ 2024-11-19 00:21:47.385967+00:00: manual__2024-11-19T00:21:47.385967+00:00, state:running, queued_at: 2024-11-19 18:29:26.120032+00:00. externally triggered: True>
[2024-11-19T19:36:46.327+0100] {logging_mixin.py:190} INFO - Failure caused by sequence item 0: expected str instance, NoneType found
[2024-11-19T19:36:46.329+0100] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=etl_pipeline_to_snowflake, task_id=run_data_transformation, run_id=manual__2024-11-19T00:21:47.385967+00:00, execution_date=20241119T002147, start_date=20241119T183645, end_date=20241119T183646
[2024-11-19T19:36:46.374+0100] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-19T19:36:46.378+0100] {standard_task_runner.py:124} ERROR - Failed to execute job 70 for task run_data_transformation (sequence item 0: expected str instance, NoneType found; 238762)
Traceback (most recent call last):
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/capstone_project/airflow/dags/scripts/transform.py", line 79, in run_transformation
    transformed_data = transform_data_s3(data)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/capstone_project/airflow/dags/scripts/transform.py", line 64, in transform_data_s3
    "Languages": df["languages"].apply(lambda x: ", ".join(x.values()) if isinstance(x, dict) else None),
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/pandas/core/series.py", line 4764, in apply
    ).apply()
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/pandas/core/apply.py", line 1209, in apply
    return self.apply_standard()
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/pandas/core/apply.py", line 1289, in apply_standard
    mapped = obj._map_values(
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/pandas/core/base.py", line 921, in _map_values
    return algorithms.map_array(arr, mapper, na_action=na_action, convert=convert)
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/airflow/airflow_coresentiment_Data_pipeline/airflow/lib/python3.10/site-packages/pandas/core/algorithms.py", line 1814, in map_array
    return lib.map_infer(values, mapper, convert=convert)
  File "lib.pyx", line 2926, in pandas._libs.lib.map_infer
  File "/mnt/c/Users/zaina/Documents/Core_Data_Engineering/capstone_project/airflow/dags/scripts/transform.py", line 64, in <lambda>
    "Languages": df["languages"].apply(lambda x: ", ".join(x.values()) if isinstance(x, dict) else None),
TypeError: sequence item 0: expected str instance, NoneType found
[2024-11-19T19:36:46.468+0100] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-11-19T19:36:46.498+0100] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-11-19T19:36:46.524+0100] {local_task_job_runner.py:245} INFO - ::endgroup::
